### PR TITLE
php client - VTCursor returns associative array rows (compatible with PDO::FETCH_BOTH)

### PR DESCRIPTION
**Pull request by Rastusik**

### CODE REVIEW

File: php/src/Vitess/StreamCursor.php [hunk 1]
```diff
@@ -1,76 +1,109 @@
...
+     */
     public function getFields()
     {
-        if ($this->fields === NULL) {
-            // The first QueryResult should have the fields.
-            if (! $this->nextQueryResult()) {
-                throw new \Vitess\Exception("stream ended before fields were received");
-            }
+        // The first QueryResult should have the fields.
+        if ($this->fields === null || !$this->nextQueryResult()) {
```
Comment (enisoc) [inline php/src/Vitess/StreamCursor.php]: This should be `&&` right?

Comment (enisoc) [inline php/src/Vitess/StreamCursor.php]: Oh you beat me to it. :)

**Comment by enisoc** [comment]
This looks great! Thanks for the incidental cleanups.

Can you just fix the VTGateConnTest so it relies on both indexed and associative entries working?

I think the simplest thing is to change [getEcho()](https://github.com/youtube/vitess/blob/master/php/tests/VTGateConnTest.php#L160) like this:

``` php
    private function getEcho($result)
    {
        $row = $result->next();
        foreach ($result->getFields() as $i => $field) {
            $this->assertEquals($row[$i], $row[$field->getName()]);
        }
        return $row;
    }
```

Then replace `self::getEcho` with `$this->getEcho`.

**Comment by Rastusik** [comment]
I'm not exactly sure how you meant it, the replaced row in your code snippet needs to stay there.

```
private function getEcho($result)
    {
        $echo = [];
        $row = $result->next();
        foreach ($result->getFields() as $i => $field) {
            $echo[$field->getName()] = $row[$i];
            $this->assertEquals($row[$i], $row[$field->getName()]);
        }
        return $echo;
    }
```

It's because there are assertions being made on the returned $echo data.

**Comment by enisoc** [comment]
Now that $row contains the values keyed by field name, it should be unnecessary to generate the $echo array, right?

**Comment by Rastusik** [comment]
Exactly. Shall I delete the method and replace the getEcho(...) calls into something like this?

```php
$result = $conn->execute($ctx, self::$ECHO_QUERY, self::$BIND_VARS, self::$TABLET_TYPE);
$row = $result->next();
... do the asserts on $row, not on $echo
```

**Comment by enisoc** [comment]
I did consider doing that, but then I realized if we do that, the test will no longer notice if the numeric keys are absent or incorrect in the returned $row (since we only assert on the string keys). That's why I favored keeping getEcho() and using it solely to check that the numeric and string keys are both present and agree with each other. Then the assertions on the string keys later will effectively check both string keys and numeric keys.

**Comment by Rastusik** [comment]
Ok, so is there anything else I should do, or is the current state sufficient? :)

**Comment by enisoc** [comment]
Sorry I wasn't clear. The current state does what I had in mind, except with a few extra steps that aren't technically necessary. It's only a test though, so it's not worth worrying about.

Thanks for the contribution!

**Pull request merged**